### PR TITLE
[sui framework] adds join_and_transfer entry fun

### LINKED ISSUE #3289: Implement `merge_and_transfer` txn
this is blocking #3287 
more contexts can be found there.

### PR DESCRIPTION
**Pull request by damirka**
Closes #3289

### CODE REVIEW

**Comment by gegaowp** [comment]
Thanks for working on this, looks great to me!
Wondering about merging a vector of Coins, is it doable? (I heard that it is not supported to take a vector of objects as argument)

**Review by gegaowp — APPROVED** [review]

File: crates/sui-framework/sources/coin.move [hunk 1]
```diff
@@ -206,6 +206,19 @@ module sui::coin {
         burn(c, coin);
     }
 
+    /// Merge two coins together.
+    public entry fun merge<T>(c1: &mut Coin<T>, c2: Coin<T>) {
+        join(c1, c2)
```
Comment (lxfind) [inline crates/sui-framework/sources/coin.move:211]: join is already an entry function, ow is this different?

**Comment by lxfind** [comment]
@gegaowp There is already `join_vec`

**Comment by lxfind** [comment]
although I am not sure if we support passing in a vector of objects at transaction level yet

**Comment by damirka** [comment]
Closing in favor of #5018

**Pull request closed without merging**